sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -165,8 +181,22 @@ case class GpuGetMapValue(child: Expression, key: Expression)
 
   override def prettyName: String = "getMapValue"
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector =
-    lhs.getBase.getMapValue(rhs.getBase)
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if (failOnError){
```

Comment:
This can be simplified as 
``` Scala
    if (failOnError){
      withResource(lhs.getBase.getMapKeyExistence(rhs.getBase)) { keyExistenceColumn =>
        withResource(keyExistenceColumn.all) { exist =>
          if (!exist.getBoolean) {
            throw new NoSuchElementException(s"Key: ${rhs.getValue.asInstanceOf[String]} does " +
              s"not exist in one of the rows in the map column")
          }
        }
      }
    }
    lhs.getBase.getMapValue(rhs.getBase)
```